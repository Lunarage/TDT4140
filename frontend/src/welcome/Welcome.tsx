import { useHistory } from "react-router-dom";
import styled from "styled-components";
import Button from "../components/Button";
import Header from "../components/Header";
import WelcomeLogo from "./WelcomeLogo";

const PageWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    background-image: url(/static/Everest.png);
    background-repeat: no-repeat;
    height: 100vh;
    background-size: auto 100%;
`;

const ContentWrapper = styled.div`
    margin: 130px 100px 10px 10px;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
`;

const WelcomeLogoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-end;
`;

const Welcome = () => {
  const history = useHistory();

  // redirecting 
  const setUrl = (url: string) => {
    history.push(url);
  };

  return (
    <PageWrapper>
      <Header />
      <ContentWrapper>
        <WelcomeLogoWrapper>
          <WelcomeLogo />
        </WelcomeLogoWrapper>
        <Button
          text="Utforsk"
          onClickFunc={() => setUrl("/browse?type=arrangementer")}
        />
        {!localStorage.getItem("token") &&
          <Button text="Logg inn" onClickFunc={() => setUrl("/login")} />}
      </ContentWrapper>
    </PageWrapper>
  );
};

export default Welcome;
