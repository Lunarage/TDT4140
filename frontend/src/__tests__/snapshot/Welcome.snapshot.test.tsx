import { mount } from "enzyme";
import Enzyme from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from "enzyme-to-json";
import Welcome from '../../welcome/Welcome';
import { store } from '../..';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual('react-router-dom') as any,
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

const WelcomeWrapper = () => (
  <Provider store={store}>
    <Welcome />
  </Provider>
);

describe("Welcome", () => {
  it("renders Welcome", () => {
    mount(<WelcomeWrapper />);
  });
  it("Should match Welcome snapshot", () => {
    const comp = mount(<WelcomeWrapper />);
    expect(toJSON(comp)).toMatchSnapshot();
  });
});

