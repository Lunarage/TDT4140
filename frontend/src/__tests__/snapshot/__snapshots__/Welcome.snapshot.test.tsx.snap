// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Should match Welcome snapshot 1`] = `
<WelcomeWrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Welcome>
      <styled.div>
        <div
          className="sc-iUuytg dteJyh"
        >
          <Header>
            <styled.div>
              <div
                className="sc-iktFzd uGybn"
              >
                <styled.div
                  onClick={[Function]}
                >
                  <div
                    className="sc-cBNfnY bDxDMJ"
                    onClick={[Function]}
                  >
                    Gjør No'
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="sc-gWHgXt bcglBl"
                  >
                    <styled.div
                      onClick={[Function]}
                    >
                      <div
                        className="sc-hiSbYr hZBClG"
                        onClick={[Function]}
                      >
                        Aktiviteter
                      </div>
                    </styled.div>
                    <styled.div
                      onClick={[Function]}
                    >
                      <div
                        className="sc-hiSbYr hZBClG"
                        onClick={[Function]}
                      >
                        Arrangementer
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
                <styled.div
                  onClick={[Function]}
                >
                  <div
                    className="sc-jJEJSO cgHETT"
                    onClick={[Function]}
                  >
                    Logg inn
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </Header>
          <styled.div>
            <div
              className="sc-iBaPrD bqRAiY"
            >
              <styled.div>
                <div
                  className="sc-eggNIi akaGd"
                >
                  <WelcomeLogo>
                    <Styled(styled.div)>
                      <div
                        className="sc-cBNfnY sc-bZSQDF bDxDMJ sFTRS"
                      >
                        Gå ut
                      </div>
                    </Styled(styled.div)>
                    <Styled(styled.div)>
                      <div
                        className="sc-cBNfnY sc-jcVebW bDxDMJ kCPxtm"
                      >
                        Gjør No'
                      </div>
                    </Styled(styled.div)>
                  </WelcomeLogo>
                </div>
              </styled.div>
              <Button
                onClickFunc={[Function]}
                text="Utforsk"
              >
                <styled.div
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff4e37",
                      "color": "white",
                      "minWidth": "8em",
                    }
                  }
                >
                  <div
                    className="sc-bdfBwQ jzdiim"
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff4e37",
                        "color": "white",
                        "minWidth": "8em",
                      }
                    }
                  >
                    <styled.div>
                      <div
                        className="sc-gsTCUz dfTHCv"
                      >
                        Utforsk
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </Button>
              <Button
                onClickFunc={[Function]}
                text="Logg inn"
              >
                <styled.div
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff4e37",
                      "color": "white",
                      "minWidth": "8em",
                    }
                  }
                >
                  <div
                    className="sc-bdfBwQ jzdiim"
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff4e37",
                        "color": "white",
                        "minWidth": "8em",
                      }
                    }
                  >
                    <styled.div>
                      <div
                        className="sc-gsTCUz dfTHCv"
                      >
                        Logg inn
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </Button>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Welcome>
  </Provider>
</WelcomeWrapper>
`;
